src/plugins/janus_sip.c

```diff
@@ -7392,6 +7419,56 @@ gpointer janus_sip_sofia_thread(gpointer user_data) {
 	return NULL;
 }
 
+/* check peer rtp has rfc2833 and push event */
+void janus_sip_check_rfc2833(janus_sip_session *session, char *buffer) {
+	janus_rtp_header *rtp_header = (janus_rtp_header *)buffer;
+	if (rtp_header->type != JANUS_RTP_RFC2833_DEFAULT_PAYLOAD_TYPE)
+		return;
+	char *payload_buffer = buffer + JANUS_RTP_HEADER_LENGTH + rtp_header->csrccount * sizeof(uint32_t);
```

Comment:
This is wrong, and doesn't skip extensions for instance, if present. There's a `janus_rtp_payload` function in `rtp.h` that does exactly all that for you. Please refer to how it's used in this and other plugins to see how to use it.

Reply:
In my last commit, I used the janus_rtp_payload function refer to how used in other plugins, but I don't know if it is correct, works fine in my case. please review it